### DISCUSSION TITLE
Rollback/Downgrade in ETCD minor and major version

### DISCUSSION THREAD

**Discussion by jaingau** [etcd-io/etcd#15843] [Q&A]
As per the ETCD documentation,  etcd support rollback in minor version like 3.5.7 to 3.5.5 or 3.5.x to 3.5.x but its not fully supported rollback from major version like 3.5.5 to 3.4.16.

So here the question,  why etcd community not supported fully rollbacks for all versions?
is there any roadmap to support full rollback in ETCD in future release?

what's need to be done to support fully rollback in ETCD?

**Comment by serathius** [comment]
Reason for lack of rollback support is simply rollbacks in distributed systems are hard.

However there is an ongoing effort on this, unfortunately no recent updates due to limited capacity of maintainers.
https://github.com/etcd-io/etcd/issues/11716 and https://github.com/etcd-io/etcd/issues/13168

Contributions are welcomed! 
Happy to help someone to onboard.
